Marlin/src/HAL/AVR/inc/SanityCheck.h

```diff
@@ -25,6 +25,30 @@
  * Test AVR-specific configuration values for errors at compile-time.
  */
 
+/**
+ * Check for common serial pin conflicts
+ */
+#define CHECK_SERIAL_PIN(N) ( \
+     X_STOP_PIN == N || Y_STOP_PIN == N || Z_STOP_PIN == N \
+  || X_MIN_PIN  == N || Y_MIN_PIN  == N || Z_MIN_PIN  == N \
+  || X_MAX_PIN  == N || Y_MAX_PIN  == N || Z_MAX_PIN  == N \
+  || X_STEP_PIN == N || Y_STEP_PIN == N || Z_STEP_PIN == N \
+  || X_DIR_PIN  == N || Y_DIR_PIN  == N || Z_DIR_PIN  == N \
+  || X_ENA_PIN  == N || Y_ENA_PIN  == N || Z_ENA_PIN  == N \
+)
+#if CONF_SERIAL_IS(0) // D0-D1. No known conflicts.
+#endif 
+#if CONF_SERIAL_IS(1) && (CHECK_SERIAL_PIN(18) || CHECK_SERIAL_PIN(19))
```

Comment:
I believe this change introduced an issue with the ATmega1284P MELZI_CREALITY board.
The ATmega1284P has its second serial port on pins D2 and D3 (Translated to pin nr 10 & 11 according to pins_MELZI_CREALITY.h).

Enabling the second  serial port causes error since pin 18 and 19 is actually used for X_STOP_PIN and Y_STOP_PIN.

Reply:
I believe your correct. 
Should be checking  D10 and D11 on 644p and 1284p

Reply:
Fixed in https://github.com/MarlinFirmware/Marlin/pull/24575

Reply:
@ellensp Wow! Thats great! 👍